source/h2_stream.c

```diff
@@ -406,6 +407,11 @@ static void s_stream_update_window(struct aws_http_stream *stream_base, size_t i
             "window size is 2147483647. We got %zu, which will cause the flow-control window to exceed the maximum",
             increment_size);
         aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+
+        struct aws_h2err returned_h2err = s_send_rst_and_close_stream(stream, aws_h2err_from_last_error());
+        if (aws_h2err_failed(returned_h2err)) {
+            aws_h2_connection_shutdown_due_to_write_err(connection, returned_h2err.aws_code);
+        }
```

Comment:
Also, part of me thinks this should ALWAYS be how we deal with window overflow:
If stream window overflows: kill the stream
If connection window overflows: kill the connection

It's much more obvious to the user how to deal with something like that. Like, I'd be cool not letting *any* of these functions return `int` because *how do you deal with these errors, as a user*? But the user ALWAYS finds out about, and has a way to deal with, connections and streams dying unexpectedly.